### DISCUSSION TITLE
Hide Sign In

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#20146] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

It would be nice to have a property in the settings file that could hide this element.

![image](https://github.com/zed-industries/zed/assets/20476002/0ad1772e-69c5-4580-8537-41e55ccb55b8)

**Comment by RemcoSmitsDev** [comment]
I think they don't want this, which makes sense: https://github.com/zed-industries/zed/pull/11802#pullrequestreview-2055715347

**Reply by robclancy** [reply]
No, it does not.

**Comment by Angelk90** [comment]
@RemcoSmitsDev : But for me and many others it makes much more sense not to have it.
If I don't want to connect Zed to my Github account now or in the future, it makes sense to me not to show it.

@maxdeviant : What's the point of showing display connection status indicator if I don't connect my github account?

**Comment by iamnbutler** [comment]
There are two things here:

- Can we remove the sign in button in the titlebar?
- Why does the no connection icon show up when I'm not signed in?

For the first, as has been mentioned above, it's unlikely we would consider removing this. Zed is collaborative software first, even if folks decide not to use those features.

For the latter, this feels incorrect. Logically, you shouldn't be able to be disconnected if you were never connected in the first place. @maxdeviant _should_ there be a time when we show this even if the user hasn't authenticated?

**Comment by Angelk90** [comment]
@iamnbutler : 
I understand that Zed started out as a workgroup editor, but how many people do you think use workgroup?
If one wants something clean and wants to use it alone, then you say don't use zed?

As you can see from the screen I like clean things, I removed everything I don't use to have as much space as possible.

Then it is simply a matter of creating a key in the settegs by setting the default value that the `Sign in` is visible and then using an if in the code that checks the value of the json variable, whether to show the writing or not.
I don't see anything so difficult it's impossible.

<img width="1440" alt="Screenshot 2024-05-28 alle 15 03 50" src="https://github.com/zed-industries/zed/assets/20476002/2aeee887-057f-49a6-b49c-26fff0246f9b">

**Comment by stormasm** [comment]
When I remove the collab features in Zed the title bar disappears completely...

https://github.com/zed-industries/zed/issues/13218#issuecomment-2308446767

**Comment by Angelk90** [comment]
I created a thread because it is agreement please vote there.

Link: #16822

**Comment by evanheller** [comment]
I also would like to cast a vote for this feature: an option to hide this "Sign In" element and its cloud connection icon. 

1. This button can be visible by default. By making this feature "opt-out," it does not alter the fundamental, collaborative nature of the software. It simply makes it an option for those users who prefer not to see this element.

2. It's debatable whether Zed is "collaborative software first." It is first and foremost a really outstanding editor that is attracting a large audience from different backgrounds. That is also has strong, well-integrated collaborative features are a strong bonus for many people, but not everyone. Why insist on the visual prominence of those features on the folks who primarily want it as "just" a really good editor?

3. In other examples of highly collaborative software on MacOS, like Sketch, Freeform, Pages, etc., the collaborative features are not always so prominent in the workspace window, and you can customize the toolbar to your liking and remove them. 

4. I can't speak to all the reasons that having a "Sign In" button visible all the time is distasteful to users, but for myself: I don't like the visual clutter unless it's something I'm using frequently, and it makes the app feel a little bit like cheap trial software. You guys have built (and are building) an awesome thing--better to keep it classy!

**Comment by mobilex1122** [comment]
I would like to have this option as well as it also annoys me. I don't mind having it on by default and have some "hidden" config in the settings that hide it if you really don't want to use it. If you want to suggest people to login you can make a "First start" popup that prompts you to log in to access more features. (With small "I don't want to log in" button)

Basically I don't mind when company tries to enforce cloud but give us option to disable it if we really don't want it. Don't be like Microsoft with the "Login to MS account" thing.

**Comment by htor** [comment]
Most things in Zed seem to be customizable so it is odd that the sign-in button should not be too. I'm usually signed in so I don't personally mind it being there, but I'm sure those who are always signed out will find it annoying to have a persistent sign-in button in the field of view.

Maybe one could have a `Configure collaboration` window that shows when a signed out user opens the collab panel, similar to `Configure Providers` when the assistant panel is opened?

**Comment by evanheller** [comment]
The developers seem really resistant to considering this option. I think it stems from a failure to appreciate that not every user of a text editor is a software developer, and such users are not thinking of collaboration and GitHub every single time they work on something. For those users, the lack of an option to hide UI elements like this Sign In button is an obvious negative.

In my case, I am a scientist, not a developer. I use text editors daily to write scripts for data analysis, to modify microscope control code and various day-to-day things like that, and to write lab protocols and manuscripts. Or just to goof around with an Arduino. Sometimes this work is collaborative in the way the developers envision, but mostly it is not. I would like Zed more if I could hide and un-hide these collaborative features on a case-by-case basis!

**Comment by Angelk90** [comment]
@evanheller : There are many discussions about this, there is this one, there are #11054, #11802, #16822 and in the future there could be many others.
But the problem does not change, the maintainers do not want the possibility of hiding the button to be given.

You can also create the PR that solves the problem see #19581, but it will never be accepted.

**Comment by stormasm** [comment]
I wrote up a simple doc basically detailing the code changes @evanheller makes in his PR #19581 

https://github.com/stormasm/zednotes/blob/main/hide-signin.md

Just for future reference as the Zed code continues to evolve this may help out down the road 😄

**Comment by kendofriendo** [comment]
This type of thing is a red flag to me. "Use the app the way we want you to." What kind of a philosophy is that?

I'm a solo dev. I don't even use github for my projects any more. Nobody's asking you to hide the button by default, or get rid of the feature entirely. We just want the _option_.

**Comment by blackhelicoptersdotnet** [comment]
To be honest, the staunch opposition to making removal of the button an *option* is a huge red flag. It gives a strong impression that a rugpull is coming, and that signing in will stop being optional. 

Zed is fantastic editor in its own right, even for those of us with no use at all for the collaboration features.

**Reply by titlestad** [reply]
I think many of us hope to be able to completely avoid the cloud. Even with AI, local is fully possible, and for us, being local only is part of a feeling of freedom while developing. Some young people may not be aware of a time where the Internet was something we connected to, it wasn't part of the computing environment. And some of us would like to keep it that way. I agree that an optional sign in is desired.

**Comment by expenses** [comment]
I'd really like to remove this button. I have absolutely zero interest in any ai, collab or sign in features. I'm using zeditor because it's fast and comfortable, that's it. Even if it's open source software, these features make it feel like it's not. If we keep going down this path I can see a fork happening.

**Comment by blepping** [comment]
I installed Zed and started configuring it. Looked for the option to remove the sign in item which I am never, ever going to use and found this discussion. Having software say "_No, you don't get to control your own machine. We make the decision for you._" is pretty much the biggest possible turnoff possible for me personally. I can deal with bugs and rough edges, but not that. Uninstalled.

**Comment by robclancy** [comment]
This is so weird, I was just configuring zed to be my other option and lost all motivation once I read this corporate ass decision. This reminds me of many other programs that refuse to remove things because of monetization and just seems like it's only a matter of time.

**Reply by blepping** [reply]
Also also such a weird, counterproductive hill to die on. I mean, you can't _force_ someone to use a feature they don't want to. The only thing you're going to do is foster distrust and ill-will. Now if anyone mentions Zed or asks for a recommendation for development environments I'm going to be bringing this up and discouraging them from using Zed.

They could just make it a hidden configuration option that doesn't even appear in the default config file. Something that only someone that was _actively_ looking to disable it would find, because that person is either not going to use it or they are going to simply give up on Zed. That still would be a little shady, but it's not going to lead to the kind of negative reaction the current approach does.

Deliberately preventing people from controlling what their own hardware/software is doing is the antithesis of open source.

**Comment by quackerex** [comment]
Even VSCode allows you to hide sign in
<img width="171" alt="Screenshot" src="https://github.com/user-attachments/assets/93c062af-294b-4cec-ac1c-3b8bb6a1879e" />

**Comment by chihabeddineAourinmouche** [comment]
Hello,

I am a developer who likes clean UI, when I look at my screen I should see EXACTLY the things I use, it's a state of mind a software developer needs to develop and that state of also covers UI and resources. Therefore, for me, a clean UI is essential for my focus.

Now addressing the excuse that "Zed is collaborative by design..."; even if I'm logged in to Zed, I do not want to see a button I am not gonna use in my UI when I don't need it. I want to see code and I want to see highlights, that's it.

I also don't want to see:
- a status bar at all times showing exactly the same info and taking up precious 1cm from the bottom of the screen
- I don't want to see a top bar showing exactly the name of the directory that I can also see right underneath it, also taking up more than 1cm off the top of the screen
- a line-number panel that takes up AN INCH??!!


Anyway, you know what for some people an unnecessary information at the top-right of the screen feels like when focused? One of those tiny insects or smudges that stick to the top-right of one of your eye-glasses...

I think refusing to address such important requests from the community is part of what gave birth to Codium in respect to VSCode.

**Reply by injust** [reply]
You can remove the line numbers -- check the `gutter` settings.

**Comment by maxdeviant** [comment]
You can now hide the sign-in button, if you so desire: https://github.com/zed-industries/zed/pull/30450

**Reply by orangecoloured** [reply]
Sorry, is there a way to also hide the connection status icon?

<img width="105" height="69" alt="Screenshot 2025-07-29 at 9 46 29 AM" src="https://github.com/user-attachments/assets/9301030a-2c6b-4809-be25-cc623662b5c9" />

**Reply by blepping** [reply]
I forget exactly which one, but one of these should do it: 

```json
  "features": {
    "edit_prediction_provider": "none"
  },
  "collaboration_panel": {
    "button": false
  },
  "agent": {
    "version": "2",
    "enabled": false
  },
  "title_bar": {
    "show_sign_in": false,
    "show_onboarding_banner": false,
    "show_user_picture": false
  },
  "toolbar": {
    "agent_review": false
  },
```

**Reply by orangecoloured** [reply]
@blepping None of these worked, but after the restart the status icon disappeared. 👌 It seems like it's bound to the ongoing session.
